docs/2-Interviews/preparing_for_interviews.md

```diff
@@ -12,22 +12,64 @@ Be prepared to:
 - talk about and work towards writing an algorithm to solve a problem, during the interview
 - you can also expect a “tell me about yourself” question (so be prepared!) and some time where they tell you about their company
 
+### What material should I know?
+
+This is a difficult question to answer. The more you know, the fewer the number of weak points that could trip you up. Knowing more things is a good thing in general, you shouldn't learn things just for interviews.
+
+That being said, you have limited time to learn things so you probably want the biggest bang for the buck.
+
+Things that interviewers will expect you to know:
+- Understanding O() analysis
+- Arrays vs linked-list
+- Stacks and queues
+- Dictionaries (one of the most used data structures in interviews)
+- Being able to traverse a binary search tree recursively.
+
+Things that are quite useful:
+- Be able to give a 5 minute answer the question "how does the internet work?" or "what happens when I type google.com in the URL bar and press enter?"
+- Breadth-first search and depth-first search on graphs
+- Being able to add and remove a node from a non-balanced binary search tree.
+- Using a heap, having an idea of how they are implemented.
+- Being able to implement quicksort and mergesort
+- Dynamic programming (most advanced concept you'll probably need to know, some companies like Google love dynamic programming questions)
+- Tries, how to use them (also relatively more advanced but they do show up not infrequently)
+
+Things that are not that important unless you have time:
+- Any graph algorithm more advanced than Djikstra search
+- Being able to implement a balanced binary search tree from scratch
+
+(The lists above are work in progress and NOT comprehensive. Submit a GitHub issue if you want to ask about a particular concept)
+
+### Know what your weaknesses are
+
+And work on them!
+
+- Have you only done assignments and/or academic work? Make sure to practice writing programs from scratch that compiles and runs. Make sure to familiarize yourself with the tools and libraries of your prefered language.
```

Comment:
Marmoset-type questions are often individual functions without IO and stuff. I think Waterloo does an ok job of making us work on some medium size projects (e.g. CS 246) but I hear some schools are even more academic and most of the work is on paper (it's true that testing real programs/executables require more resources).